txscript/script.go

```diff
@@ -385,7 +385,6 @@ func calcSignatureHash(script []parsedOpcode, hashType SigHashType,
 	}
 	if hashType&SigHashAnyOneCanPay != 0 {
 		txCopy.TxIn = txCopy.TxIn[idx : idx+1]
-		idx = 0
```

Comment:
In this case, I think we should comment it.  This is quite important if the code is changed in the future to do something with the idx after and would be super easy to forget.

Reply:
Not sure why github shows this still...  it was commented on.